pandas/plotting/_matplotlib/core.py

```diff
@@ -413,7 +414,13 @@ def _compute_plot_data(self):
         # np.ndarray before plot.
         numeric_data = numeric_data.copy()
         for col in numeric_data:
-            numeric_data[col] = np.asarray(numeric_data[col])
+
+            # GH32073: cast to float if values contain nulled integers
+            values = numeric_data[col]
+            if values.isna().any().all() and is_integer_dtype(values.dtype):
```

Comment:
Cheaper to check the dtype rather than the validity of each value.

```suggestion
            if is_integer_dtype(values.dtype) and is_extension_array_dtype(values.dtype):
```

Do we know that `values` is 1-D here?

Reply:
Although there is no clear checking, `values` comes from `numeric_data[col]`, so I don't think we would have something different than 1-D here.... I ran tests locally, and it should work.